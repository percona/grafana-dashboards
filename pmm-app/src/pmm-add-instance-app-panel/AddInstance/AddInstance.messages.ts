export const Messages = {
  form: {
    trackingOptions: {
      none: 'Don\'t track',
      pgStatements: 'PG Stat Statements',
      pgMonitor: 'PG Stat Monitor',
    },
    labels: {
      trackingOptions: 'Stat tracking options',
    },
  },
};
