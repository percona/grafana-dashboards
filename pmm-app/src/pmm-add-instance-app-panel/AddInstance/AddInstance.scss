#antd {
  .content-wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .navigation {
    display: flex;
    flex-direction: column;
    max-width: 550px;
    width: 100%;
    overflow: hidden;
    border-radius: 0.25rem;
    background-color: var(--highlight-background);
  }

  .navigation__link {
    padding: 1rem;
    color: inherit;
  }

  .navigation__link:hover {
    background-color: var(--navigation-link-hover);
    text-decoration: none;
    color: inherit;
  }

  .navigation__link:not(:last-child) {
    border-bottom: 1px solid rgba(0, 0, 0, 0.125);
  }
}
