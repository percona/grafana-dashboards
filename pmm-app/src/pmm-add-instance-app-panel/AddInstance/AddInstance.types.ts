export enum TrackingOptions {
  none = '',
  pgStatements = 'qan_postgresql_pgstatements_agent',
  pgMonitor = 'qan_postgresql_pgstatmonitor_agent',
}
