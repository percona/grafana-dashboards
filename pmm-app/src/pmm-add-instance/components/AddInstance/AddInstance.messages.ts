export const Messages = {
  titles: {
    rds: 'AWS RDS MySQL or Aurora MySQL',
    postgresql: 'PostgreSQL',
    mysql: 'MySQL',
    mongodb: 'MongoDB',
    proxysql: 'ProxySQL',
    addInstance: 'Add a remote instance',
  },
};
