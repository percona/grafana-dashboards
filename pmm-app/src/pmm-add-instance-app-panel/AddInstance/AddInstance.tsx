import React from 'react';
import './AddInstance.scss';

const AddInstance = props => {
  const selectInstanceType = type => () => {
    props.onSelectInstanceType({ type });
  };
  return (
    <section className="content-wrapper">
      <h3>How to Add an Instance</h3>
      <nav className="navigation">
        <a href="#" className="navigation__link" onClick={selectInstanceType('rds')}>
          <i className="fa fa-pencil" aria-hidden="true" /> Add an AWS RDS MySQL or Aurora MySQL Instance
        </a>
        <a href="#" className="navigation__link" onClick={selectInstanceType('postgresql')}>
          <i className="fa fa-pencil" aria-hidden="true" /> Add a Remote PostgreSQL Instance
        </a>
        <a href="#" className="navigation__link" onClick={selectInstanceType('mysql')}>
          <i className="fa fa-pencil" aria-hidden="true" /> Add a Remote MySQL Instance
        </a>
        <a href="#" className="navigation__link" onClick={selectInstanceType('mongodb')}>
          <i className="fa fa-pencil" aria-hidden="true" /> Add a Remote MongoDB Instance
        </a>
        <a href="#" className="navigation__link" onClick={selectInstanceType('proxysql')}>
          <i className="fa fa-pencil" aria-hidden="true" /> Add a Remote ProxySQL Instance
        </a>
      </nav>
    </section>
  );
};

export default AddInstance;
