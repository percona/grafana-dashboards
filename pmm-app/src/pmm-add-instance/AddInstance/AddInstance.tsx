import React, { Component } from "react";
import "./AddInstance.scss";

class AddInstance extends Component {
  render() {
    return (
      <section className="content-wrapper">
        <h3>How to Add an Instance</h3>
        <nav className="navigation">
          <a href="/add-remote-postgresql" className="navigation__link">
            <i className="fa fa-pencil" aria-hidden="true"></i> Add a Remote
            PostgreSQL Instance
          </a>
          <a href="/add-remote-mysql" className="navigation__link">
            <i className="fa fa-pencil" aria-hidden="true"></i> Add a Remote
            MySQL Instance
          </a>
          <a href="/add-remote-mongodb" className="navigation__link">
            <i className="fa fa-pencil" aria-hidden="true"></i> Add a Remote
            MongoDB Instance
          </a>
          <a href="/add-remote-proxysql" className="navigation__link">
            <i className="fa fa-pencil" aria-hidden="true"></i> Add a Remote
            ProxySQL Instance
          </a>
        </nav>
      </section>
    );
  }
}

export default AddInstance;
