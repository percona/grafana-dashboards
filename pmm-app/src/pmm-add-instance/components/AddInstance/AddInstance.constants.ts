import { Messages } from './AddInstance.messages';
import { InstanceTypes } from '../../panel.types';

export const instanceList = [
  { type: InstanceTypes.rds, title: Messages.titles.rds },
  { type: InstanceTypes.postgresql, title: Messages.titles.postgresql },
  { type: InstanceTypes.mysql, title: Messages.titles.mysql },
  { type: InstanceTypes.mongodb, title: Messages.titles.mongodb },
  { type: InstanceTypes.proxysql, title: Messages.titles.proxysql },
  { type: InstanceTypes.external, title: Messages.titles.external },
];
