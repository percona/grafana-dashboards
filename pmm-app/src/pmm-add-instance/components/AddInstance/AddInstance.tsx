import React, { FC } from 'react';
import { useTheme } from '@grafana/ui';
import { Database } from 'shared/components/Elements/Icons/Database';
import { getStyles } from './AddInstance.styles';
import { instanceList } from './AddInstance.constants';
import { Messages } from './AddInstance.messages';
import { AddInstanceProps, SelectInstanceProps } from './AddInstance.types';

export const SelectInstance: FC<SelectInstanceProps> = ({ type, selectInstanceType, title }) => {
  const theme = useTheme();
  const styles = getStyles(theme);

  return (
    <button
      className={styles.navigationButton}
      data-qa={`${type}-instance`}
      onClick={selectInstanceType(type)}
      type="button"
    >
      <Database />
      <span className={styles.addInstanceTitle}>
        {title}
      </span>
      <span className={styles.addInstance}>{Messages.titles.addInstance}</span>
    </button>
  );
};

export const AddInstance: FC<AddInstanceProps> = ({ onSelectInstanceType }) => {
  const theme = useTheme();
  const styles = getStyles(theme);

  const selectInstanceType = (type) => () => onSelectInstanceType({ type });

  return (
    <section className={styles.content}>
      <nav className={styles.navigationPanel}>
        {instanceList.map((item) => (
          <SelectInstance
            selectInstanceType={selectInstanceType}
            type={item.type}
            title={item.title}
            key={item.type}
          />
        ))}
      </nav>
    </section>
  );
};
