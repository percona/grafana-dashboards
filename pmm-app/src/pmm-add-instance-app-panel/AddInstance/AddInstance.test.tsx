// Just a stub test
import AddInstance from './AddInstance';
import React from 'react';
import renderer from 'react-test-renderer';

xdescribe('AddInstance page render test', () => {
  it('AddInstance renders correct without props', () => {
    const component = renderer.create(<AddInstance />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
