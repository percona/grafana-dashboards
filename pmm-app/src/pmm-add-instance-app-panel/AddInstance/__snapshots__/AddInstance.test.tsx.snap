// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddInstance page AddInstance renders correct without props 1`] = `
<section
  className="content-wrapper"
>
  <h3>
    How to Add an Instance
  </h3>
  <nav
    className="navigation"
  >
    <a
      className="navigation__link"
      href="#"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-pencil"
      />
       
      Add an AWS RDS MySQL or Aurora MySQL Instance
    </a>
    <a
      className="navigation__link"
      href="#"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-pencil"
      />
       
      Add a Remote PostgreSQL Instance
    </a>
    <a
      className="navigation__link"
      href="#"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-pencil"
      />
       
      Add a Remote MySQL Instance
    </a>
    <a
      className="navigation__link"
      href="#"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-pencil"
      />
       
      Add a Remote MongoDB Instance
    </a>
    <a
      className="navigation__link"
      href="#"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-pencil"
      />
       
      Add a Remote ProxySQL Instance
    </a>
  </nav>
</section>
`;
